app/controllers/application_controller.rb

```diff
@@ -23,10 +25,45 @@ class ApplicationController < ActionController::Base
     )
   end
 
+  class << self
+  protected
+
+    def force_html_format(only: nil, except: nil)
+      only_copy = only && only.collect(&:to_sym).freeze
+      except_copy = except && except.collect(&:to_sym).freeze
+      define_method(:force_html_format_only) { only_copy }
+      define_method(:force_html_format_except) { except_copy }
+    end
```

Comment:
I can't follow what this code is doing Mat sorry, can you make it more verbose so it's easier to follow?

I'm also wondering more broadly if this level of solution is necessary. I'm wondering if we can do something simpler like having a before action in the application controller to check if the `request.format.js?` and if it is, handling that with a suitable error. 

I'm tagging @KludgeKML here for his opinion as well

Reply:
Hi. I'll improve the code in the final version. Let me just explain it here now. It created two methods, the first one containing an array of controller actions that should be forced to html (only), the second one containing an array of controller action which should be excepted from being forced to html (except).

With respect to the further issues you mentioned, the bot makes js requests, so if you are ok with maybe returning 415 (Unsupported Media Type) for all non-html requests, should I do that instead of what is suggested in this PR?

Reply:
@hannako @KludgeKML I've created another PR containing Jess' solution. I think that's better than this one, so I'm closing this PR.